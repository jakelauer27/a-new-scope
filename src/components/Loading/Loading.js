import React from 'react'

export default function Loading() {
  return (
    <div className='loading-container'>
      <div className='loading-animation'>
        <div className='millenium-div'>
          <img className='loading-millenium' src='./images/load_millenium.svg' alt='millenium falcon'></img>
          <div className='blaster1 blaster'></div>
          <div className='blaster2 blaster'></div>
        </div>
        <div className='tie-div'>
          <img className='loading-tiefighter' src='./images/load_tiefighter.svg' alt='millenium falcon'></img>
        </div>
      </div>
      <h2 className='loading-text'>Loading...</h2>
    </div>
  )
}