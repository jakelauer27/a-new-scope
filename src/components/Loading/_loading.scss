.loading-text {
  animation: flashing 2s linear 0s infinite alternate;
  color: silver;
  font-family: 'Pathway Gothic One', sans-serif;
  letter-spacing: 6px;
  font-size: 2.5em;
  position: absolute;
  bottom: 15px;
  margin: 0px;
  right: 22px;
}

.loading-animation {
  animation: rotateRing 20s linear 0s infinite;
  display: flex;
  height: 300px;
  margin: calc(50vh - 150px) auto;
  justify-content: space-between;
  position: relative;
  top: 0px;
  left: -100px;
  width: 300px;
}

.millenium-div {
  animation: loadMill 5s ease-in-out 0s infinite;
  width: 0px;
  height: 0px;
}

.loading-millenium {
  width: 120px;
}

.loading-tiefighter {
  animation: loadTie 5s ease-in-out 0s infinite;
  width: 70px;
}

.blaster {
  animation: blaster .3s linear 0s infinite;
  border-radius: 2px;
  background-color: rgb(72, 255, 0);
  box-shadow: 0 0 2px 1px rgb(72, 255, 0),
  0 0 15px 2px rgb(72, 255, 0),
  0 0 25px 1px rgb(72, 255, 0),
inset 0 0 2px 1px rgb(72, 255, 0),
inset 0 0 25px 1px rgb(72, 255, 0),
inset 0 0 15px 2px rgb(72, 255, 0);
  position: relative;
}

.blaster1 {
  height: 4px;
  width: 9px;
  top: -98px;
  right: -25px;
}

.blaster2 {
  height: 3px;
  width: 11px;
  top: -68px;
  right: -26px;
}

@keyframes loadMill {
  0% {
    transform: 
      rotate(180deg);
  }

  50% {
    transform: 
      translateX(200px)
      rotate(0deg)
  }

  100% {
    transform: 
      translateX(0px)
      rotate(180deg)
  }
}

@keyframes loadTie {
  0% {
    transform: 
      rotate(180deg);
  }

  50% {
    transform: 
      translateX(-200px)
      rotate(0deg)
  }

  100% {
    transform: 
      translateX(0px)
      rotate(180deg)
  }
}

@keyframes blaster {
  0% {
    transform: translateX(0px)
  }
  50% {
    transform: translateX(-400px)
  }
  51% {
    opacity: 0;
  }
  100% {
    opacity: 0;
  }
}

@keyframes flashing {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}