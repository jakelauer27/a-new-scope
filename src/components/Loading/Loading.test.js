import React from 'react'
import { shallow } from 'enzyme'
import Loading from './Loading'

describe('Loading Component', () => {
  it('Should match the snapshot', () => {
    const wrapper = shallow(<Loading />)

    expect(wrapper).toMatchSnapshot()
  })
})
