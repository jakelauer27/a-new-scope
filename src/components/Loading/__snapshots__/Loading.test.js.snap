// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="loading-animation"
        >
          <div
            className="millenium-div"
          >
            <img
              alt="millenium falcon"
              className="loading-millenium"
              src="./images/load_millenium.svg"
            />
            <div
              className="blaster1 blaster"
            />
            <div
              className="blaster2 blaster"
            />
          </div>
          <div
            className="tie-div"
          >
            <img
              alt="millenium falcon"
              className="loading-tiefighter"
              src="./images/load_tiefighter.svg"
            />
          </div>
        </div>,
        <h2
          className="loading-text"
        >
          Loading...
        </h2>,
      ],
      "className": "loading-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="millenium-div"
            >
              <img
                alt="millenium falcon"
                className="loading-millenium"
                src="./images/load_millenium.svg"
              />
              <div
                className="blaster1 blaster"
              />
              <div
                className="blaster2 blaster"
              />
            </div>,
            <div
              className="tie-div"
            >
              <img
                alt="millenium falcon"
                className="loading-tiefighter"
                src="./images/load_tiefighter.svg"
              />
            </div>,
          ],
          "className": "loading-animation",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="millenium falcon"
                  className="loading-millenium"
                  src="./images/load_millenium.svg"
                />,
                <div
                  className="blaster1 blaster"
                />,
                <div
                  className="blaster2 blaster"
                />,
              ],
              "className": "millenium-div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "millenium falcon",
                  "className": "loading-millenium",
                  "src": "./images/load_millenium.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "blaster1 blaster",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "blaster2 blaster",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="millenium falcon"
                className="loading-tiefighter"
                src="./images/load_tiefighter.svg"
              />,
              "className": "tie-div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "millenium falcon",
                "className": "loading-tiefighter",
                "src": "./images/load_tiefighter.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
          "className": "loading-text",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="loading-animation"
          >
            <div
              className="millenium-div"
            >
              <img
                alt="millenium falcon"
                className="loading-millenium"
                src="./images/load_millenium.svg"
              />
              <div
                className="blaster1 blaster"
              />
              <div
                className="blaster2 blaster"
              />
            </div>
            <div
              className="tie-div"
            >
              <img
                alt="millenium falcon"
                className="loading-tiefighter"
                src="./images/load_tiefighter.svg"
              />
            </div>
          </div>,
          <h2
            className="loading-text"
          >
            Loading...
          </h2>,
        ],
        "className": "loading-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="millenium-div"
              >
                <img
                  alt="millenium falcon"
                  className="loading-millenium"
                  src="./images/load_millenium.svg"
                />
                <div
                  className="blaster1 blaster"
                />
                <div
                  className="blaster2 blaster"
                />
              </div>,
              <div
                className="tie-div"
              >
                <img
                  alt="millenium falcon"
                  className="loading-tiefighter"
                  src="./images/load_tiefighter.svg"
                />
              </div>,
            ],
            "className": "loading-animation",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="millenium falcon"
                    className="loading-millenium"
                    src="./images/load_millenium.svg"
                  />,
                  <div
                    className="blaster1 blaster"
                  />,
                  <div
                    className="blaster2 blaster"
                  />,
                ],
                "className": "millenium-div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "millenium falcon",
                    "className": "loading-millenium",
                    "src": "./images/load_millenium.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "blaster1 blaster",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "blaster2 blaster",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="millenium falcon"
                  className="loading-tiefighter"
                  src="./images/load_tiefighter.svg"
                />,
                "className": "tie-div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "millenium falcon",
                  "className": "loading-tiefighter",
                  "src": "./images/load_tiefighter.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
            "className": "loading-text",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
